fe/fe-core/src/main/java/com/starrocks/sql/optimizer/GroupExpression.java

```diff
@@ -35,11 +38,15 @@
     private final BitSet ruleMasks = new BitSet(RuleType.NUM_RULES.ordinal() + 1);
     private boolean statsDerived = false;
     private final Map<PhysicalPropertySet, Pair<Double, List<PhysicalPropertySet>>> lowestCostTable;
+    private final Set<Pair<PhysicalPropertySet, List<PhysicalPropertySet>>> validOutputInputProperties;
+    private Map<Pair<PhysicalPropertySet, List<PhysicalPropertySet>>, Integer> propertiesPlanCountMap;
 
     public GroupExpression(Operator op, List<Group> inputs) {
         this.op = op;
         this.inputs = inputs;
         this.lowestCostTable = Maps.newHashMap();
+        this.validOutputInputProperties = Sets.newLinkedHashSet();
+        this.propertiesPlanCountMap = Maps.newLinkedHashMap();
```

Comment:
It's need keep order?

Reply:
Yes，in order to get same outputProperty/inputProperty with same nTH